app/src/scripts/common/partials/spinner.jsx

```diff
@@ -6,27 +6,55 @@ import PropTypes from 'prop-types'
 // component setup ---------------------------------------------------------------------------
 
 export default class Spinner extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = { timedOut: false }
+  }
   // life cycle events -------------------------------------------------------------------------
+  componentDidMount() {
+    this._isMounted = true
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false
+  }
 
   render() {
-    let spinner = (
-      <div className="loading-wrap fade-in">
-        <div className="spinner">
-          <div className="spinnerinner" />
+    if (!this.state.timedOut) {
+      if (this.props.timeout) {
+        setTimeout(() => {
```

Comment:
It's better to do this when the component mounts, instead of checking in render for the mounted state. componentWillReceiveProps is called on the prop change when the component is mounted.

Reply:
It also should call clearTimeout when the component is unmounted to avoid stacking timeouts. It would be better to create a new timeout component and wrap the spinner in it, that way if we fix the components that don't bubble up errors later, we can remove the extra component without changing the spinner again.